sdk/cosmosdb/cosmos/src/auth.ts

```diff
@@ -62,6 +62,17 @@ export async function setAuthorizationHeader(
     headers[Constants.HttpHeaders.Authorization] = encodeURIComponent(
       await clientOptions.tokenProvider({ verb, path, resourceId, resourceType, headers })
     );
+  } else if (clientOptions.aadCredentials) {
```

Comment:
Cosmos Authentication (master key etc...) is not retried on failures. Where as AAD on token expiry needs to be retired, how is it handled?

Reply:
found it.